utilities/responses.py

```diff
@@ -2,44 +2,42 @@
 main response handler, some of response handel throw  middleware
 """
 import time
-import json
 
-from django.http import HttpResponse
 from django.conf import settings
 from django.utils.translation import ugettext as _
+from rest_framework.response import Response
 
 
 class BaseResponse:
+    def __init__(self, **kwargs):
+        self.message = kwargs.get('message')
+        self.show_type = kwargs.get('show_type', settings.MESSAGE_SHOW_TYPE['TOAST'])
+        self.current_time = round(time.time())
+        self.status = kwargs.get("status")
+        self.success = kwargs.get("success")
+
     def send(self):
         status = self.__dict__.pop('status')
-        return HttpResponse(
-            json.dumps(self.__dict__),
-            status=status,
-            content_type="application/json"
+        return Response(
+            data=self.__dict__,
+            status=status
         )
 
 
 class ErrorResponse(BaseResponse):
-    def __init__(self, message, dev_error=None, errors=None, show_type=settings.MESSAGE_SHOW_TYPE['TOAST'], status=400):
-        self.message = message
+    def __init__(self, dev_error=None, errors=None, **kwargs):
+        super().__init__(**kwargs)
         self.dev_error = dev_error
         self.errors = errors
-        self.show_type = show_type
-        self.current_time = round(time.time())
-        self.success = False
-        self.status = status
 
 
 class SuccessResponse(BaseResponse):
-    def __init__(self, data=None, message=None, show_type=settings.MESSAGE_SHOW_TYPE['TOAST'], status=200, **kwargs):
-        self.data = data
+    def __init__(self, message, data=None, **kwargs):
```

Comment:
Message could be None

Reply:
You are right. I get it wrong with ErrorResponse which message should be positional.
Also status should be checked to it's not be None.